File: app_dev/frontend/src/lib/roles.ts
// Espelha app_dev/backend/app/domains/users/roles.py — mantido em sincronia manual.

export interface UserItem {
  id: number;
  email: string;
  nome: string;
  role: string;
  ativo: number;
}

/**
 * Perfis de acesso. `atendimento` é restrito: entra apenas em
 * atendimento.atelieilmaguerra.com.br, e não vê financeiro, plano nem parâmetros.
 */
export const ROLES = [
  { valor: "user", rotulo: "Usuário", curto: "Usuário", sub: "Acesso ao sistema de gestão" },
  { valor: "admin", rotulo: "Administrador", curto: "Admin", sub: "Acesso total, incluindo usuários" },
  {
    valor: "atendimento",
    rotulo: "Atendimento",
    curto: "Atendimento",
    sub: "Só o site de atendimento: clientes e pedidos para aprovação",
  },
];

export function rotuloRole(role: string, curto = false): string {
  const r = ROLES.find((x) => x.valor === role);
  if (!r) return role;
  return curto ? r.curto : r.rotulo;
}

File: app_atendimento/backend/app/domains/users/roles.py
"""
Perfis de acesso.

O ateliê tem dois sites com bases parcialmente compartilhadas:

- gestão (este app)         — acesso total: pedidos, financeiro, plano, despesas
- atendimento (:8002)       — só clientes e pedidos aguardando aprovação

`atendimento` é uma role de acesso restrito: entra apenas no site de atendimento.
`admin` entra nos dois. `user` só na gestão.

ESPELHO de app_dev/backend/app/domains/users/roles.py.
Cópia literal — se mexer lá, mexa aqui. O site de atendimento é um app separado
de propósito (isolamento de dados), e o preço disso é manter estes poucos
arquivos em sincronia.
"""

ROLE_ADMIN = "admin"
ROLE_USER = "user"
ROLE_ATENDIMENTO = "atendimento"

ROLES_VALIDAS = (ROLE_ADMIN, ROLE_USER, ROLE_ATENDIMENTO)

# Quem pode fazer login no app de gestão.
ROLES_GESTAO = (ROLE_ADMIN, ROLE_USER)

# Quem pode fazer login no app de atendimento (espelhado em app_atendimento).
ROLES_ATENDIMENTO = (ROLE_ADMIN, ROLE_ATENDIMENTO)

ROLE_LABELS = {
    ROLE_ADMIN: "Administrador",
    ROLE_USER: "Usuário",
    ROLE_ATENDIMENTO: "Atendimento",
}
